shared/src/db/database.ts

```diff
@@ -626,6 +630,51 @@ class database extends Dexie {
             });
     }
 
+    private deleteExpired() {
+        console.log(`isCMS: ${cmsFlag.cms}`);
+
+        if (cmsFlag.cms) {
+            console.log("CMS is on");
+            return;
+        } else {
+            console.log("CMS is not on...deleting documents");
+            const now = DateTime.now().toMillis();
+
+            this.docs
+                .where("type")
+                .anyOf(DocType.Content, DocType.Post)
+                .toArray()
+                .then((docs: BaseDocumentDto[]) => {
+                    const expiredDocs = docs.filter((doc) => {
+                        if (doc.type === DocType.Content) {
+                            const contentDoc = doc as ContentDto;
+                            return contentDoc.expiryDate && contentDoc.expiryDate <= now;
+                        } else if (doc.type === DocType.Post) {
+                            const postDoc = doc as PostDto;
+                            if (postDoc.metadata && postDoc.metadata.length > 0) {
+                                for (const metadata of postDoc.metadata) {
+                                    if (metadata.expiryDate && metadata.expiryDate <= now) {
+                                        return true; // Expired document found
+                                    }
+                                }
```

Comment:
I _think_ there is no `metadata` field on the post, because the expiry dates are per content, not globally for the whole post.